modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    private Bundle getBundledConstants() {
+        Bundle constants = new Bundle();
+
+        ArrayList<Locale> locales = getLocales();
+        ArrayList<String> localeNames = getLocaleNames(locales);
+        Boolean isRTL = TextUtils.getLayoutDirectionFromLocale(Locale.getDefault())== View.LAYOUT_DIRECTION_RTL;
+
+        constants.putBoolean("isRTL", isRTL);
+        constants.putString("locale", localeNames.get(0));
+        constants.putStringArrayList("locales", localeNames);
+        constants.putString("timezone", TimeZone.getDefault().getID());
+        constants.putStringArrayList("isoCurrencyCodes", getISOCurrencyCodes());
+        constants.putString("country", locales.get(0).getCountry());
+
+        return constants;
+    }
+
+    private ArrayList<String> getISOCurrencyCodes() {
+        ArrayList<String> locales = new ArrayList<>();
+        final Set<Currency> availableCurrencies = getAvailableCurrencies();
+        for (Currency handle : availableCurrencies) locales.add(handle.getCurrencyCode());
+        return locales;
+    }
+
+    private ArrayList<Locale> getLocales() {
+        ArrayList<Locale> locales = new ArrayList<>();
+
+        Configuration configuration = getApplicationContext().getResources().getConfiguration();
+        if (SDK_INT >= N) {
+            LocaleList localeList = configuration.getLocales();
+            for (int i = 0; i < localeList.size(); i++)
+                locales.add(localeList.get(i));
+        } else {
+            locales.add(configuration.locale);
+        }
+
+        return locales;
+    }
+
+    private ArrayList<String> getLocaleNames(ArrayList<Locale> locales) {
+        ArrayList<String> localeNames = new ArrayList<>();
+        for (int i = 0; i < locales.size(); i++) localeNames.add(toLocaleTag(locales.get(i)));
+        return localeNames;
+    }
+
+    private String toLocaleTag(Locale locale) {
+        String localeTag;
+
+        if (SDK_INT >= LOLLIPOP) {
+            localeTag = locale.toLanguageTag();
+        } else {
+            StringBuilder builder = new StringBuilder();
+            builder.append(locale.getLanguage());
+            if (locale.getCountry() != null) {
+                builder.append("-");
+                builder.append(locale.getCountry());
+            }
+            localeTag = builder.toString();
+        }
+
+        if (localeTag.matches("^(iw|in|ji).*")) {
+            localeTag = localeTag
+                    .replace("iw", "he")
+                    .replace("in", "id")
+                    .replace("ji", "yi");
+        }
+
+        return localeTag;
+    }
+
+    private void onLocaleUpdated() {
+        mEventEmitter.emit("Expo.onLocaleUpdated", getBundledConstants());
+    }
+
+    private class LocalesBroadcastReceiver extends BroadcastReceiver {
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction().equals(ACTION_LOCALE_CHANGED)) onLocaleUpdated();
```

Comment:
I think this receiver should hold a WeakReference to `LocalizationModule.this` so that `LocalizationModule` isn't strongly referenced by the receiver.

If the experience closes, `setModuleRegistry` isn't called with `null` module registry, but the MR just removes a strong reference to the module. So I think we should structure `LocalesBroadcastReceiver` in such way that it

1. removes itself if `LocalizationModule.this` doesn't exist anymore
2. doesn't reference `LocalizationModule` by a strong reference